### DISCUSSION TITLE
About "UploadFile"

### DISCUSSION THREAD

**Discussion by FuckAnyProblem** [fastapi/fastapi#6147] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@PostRouter.post("/file", response_model=Response)
async def UploadSmall(request: Request,
                      document: UploadFile = File(..., alias="document"),
                      password: Union[str, None] = Body(default=None,
                                                        min_length=4,
                                                        max_length=11,
                                                        alias="password")):
    headers = request.headers
    cl = int(headers["content-length"])
    if cl > int(Configs["FILE"]["MAXSIZE"]):
        return FileUploadFail(Msg="超过最大文件限制大小")
    if document.content_type not in allow_type:
        return FileUploadFail(Msg="文件类型不允许")
    name = document.filename.rsplit(".", maxsplit=1)[0]
    if len(await SSR.SelectAllKeys("UPLOAD." + name)) > 0:
        return FileUploadFail(Msg="重复文件名")
    l = document.file.read()
    # Attention: There is a value pass
    md5 = Md5Hex(l)
    print(name, md5)
    ftype = document.filename.split(".")[-1]
    if await SaveFile("./statics/" + md5 + "." + ftype, l) is False:
        return FileUploadFail(Msg="未知错误，请联系开发者")
    temp = {}
    temp["md5"] = md5
    temp["date"] = Now()
    temp["size"] = str(round(cl / 1024 / 1024, 2)) + "MB"
    temp["type"] = ftype
    if password:
        temp["password"] = password
    await SSR.SetKeyValue("UPLOAD." + name, json.dumps(temp))
    await SSR.SetKeyValue("FILE." + temp["md5"],
                          "./statics/" + md5 + "." + temp["type"])
    return FileUploadSuccess(Data=SSFile(
        Name=name,
        Date=temp["date"],
        Size=temp["size"],
        Type=temp["type"],
    ))
```

### Description

I'am uploading pics, but one pic cant be uploaded, other pics can be uploaded.

Here is the  error pic
![dasdas](https://user-images.githubusercontent.com/114020110/194763592-5c1b873d-380c-41df-95c1-8d3fe79896af.png)

### Operating System

Linux

### Operating System Details

Ubuntu 20.04

### FastAPI Version

0.85.0

### Python Version

Python3.8.2

### Additional Context

![image](https://user-images.githubusercontent.com/114020110/194763478-ac3a6b5a-365a-4af3-a455-b546b51b838b.png)

**Comment by FuckAnyProblem** [comment]
> ### First Check
> * [x]  I added a very descriptive title to this issue.
> * [x]  I used the GitHub search to find a similar issue and didn't find it.
> * [x]  I searched the FastAPI documentation, with the integrated search.
> * [x]  I already searched in Google "How to X in FastAPI" and didn't find any information.
> * [x]  I already read and followed all the tutorial in the docs and didn't find an answer.
> * [x]  I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
> * [x]  I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
> * [x]  I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
> 
> ### Commit to Help
> * [x]  I commit to help with one of those options 👆
> 
> ### Example Code
> ```python
> @PostRouter.post("/file", response_model=Response)
> async def UploadSmall(request: Request,
>                       document: UploadFile = File(..., alias="document"),
>                       password: Union[str, None] = Body(default=None,
>                                                         min_length=4,
>                                                         max_length=11,
>                                                         alias="password")):
>     headers = request.headers
>     cl = int(headers["content-length"])
>     if cl > int(Configs["FILE"]["MAXSIZE"]):
>         return FileUploadFail(Msg="超过最大文件限制大小")
>     if document.content_type not in allow_type:
>         return FileUploadFail(Msg="文件类型不允许")
>     name = document.filename.rsplit(".", maxsplit=1)[0]
>     if len(await SSR.SelectAllKeys("UPLOAD." + name)) > 0:
>         return FileUploadFail(Msg="重复文件名")
>     l = document.file.read()
>     # Attention: There is a value pass
>     md5 = Md5Hex(l)
>     print(name, md5)
>     ftype = document.filename.split(".")[-1]
>     if await SaveFile("./statics/" + md5 + "." + ftype, l) is False:
>         return FileUploadFail(Msg="未知错误，请联系开发者")
>     temp = {}
>     temp["md5"] = md5
>     temp["date"] = Now()
>     temp["size"] = str(round(cl / 1024 / 1024, 2)) + "MB"
>     temp["type"] = ftype
>     if password:
>         temp["password"] = password
>     await SSR.SetKeyValue("UPLOAD." + name, json.dumps(temp))
>     await SSR.SetKeyValue("FILE." + temp["md5"],
>                           "./statics/" + md5 + "." + temp["type"])
>     return FileUploadSuccess(Data=SSFile(
>         Name=name,
>         Date=temp["date"],
>         Size=temp["size"],
>         Type=temp["type"],
>     ))
> ```
> 
> ### Description
> I'am uploading pics, but one pic cant be uploaded, other pics can be uploaded.
> 
> Here is the error pic ![dasdas](https://user-images.githubusercontent.com/114020110/194763592-5c1b873d-380c-41df-95c1-8d3fe79896af.png)
> 
> ### Operating System
> Linux
> 
> ### Operating System Details
> Ubuntu 20.04
> 
> ### FastAPI Version
> 0.85.0
> 
> ### Python Version
> Python3.8.2
> 
> ### Additional Context
> ![image](https://user-images.githubusercontent.com/114020110/194763478-ac3a6b5a-365a-4af3-a455-b546b51b838b.png)

I have already find solution in stackflow, but it doesn't work. here [https://stackoverflow.com/questions/62418392/size-of-pdf-breaks-fastapi-using-python-multipart](url)

**Comment by tiangolo** [comment, marked as answer]
I assume you solved it, thanks for closing the issue 👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.

**Comment by FuckAnyProblem** [comment]
> I assume you solved it, thanks for closing the issue 👍
> 
> > Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.

Thanks for your reply.
Yeah, It's postman's bad.
